pennylane/queuing.py

```diff
@@ -173,6 +173,22 @@ class QueuingError(Exception):
     """Exception that is raised when there is a queuing error"""
 
 
+class WrappedObj:
+    """Wraps an object to make its has dependent on its identity"""
+
+    def __init__(self, obj):
+        self.obj = obj
+
+    def __hash__(self):
+        return id(self.obj)
+
+    def __eq__(self, other):
+        return id(self.obj) == id(other.obj)
```

Comment:
Is there any advantage to storing the id as an attribute? This way we don't have to call `hash` multiple times?

Reply:
I don't think the speedup would be significant enough to matter, and storing the id as an attribute makes the code a bit less readable in my opinion.

Reply:
Sounds good 👍🏼